Revit_Core_Engine/Convert/Geometry/FromRevit/Surface.cs

```diff
@@ -43,9 +43,9 @@ public static oM.Geometry.PlanarSurface FromRevit(this PlanarFace face)
             if (face == null)
                 return null;
 
-            IList<CurveLoop> crvLoop = face.GetEdgesAsCurveLoops();
-            oM.Geometry.ICurve externalBoundary = crvLoop[0].FromRevit();
-            List<oM.Geometry.ICurve> internalBoundary = crvLoop.Skip(1).Select(x => x.FromRevit() as oM.Geometry.ICurve).ToList();                           
+            List<CurveLoop> crvLoops = face.GetEdgesAsCurveLoops().ToList();
+            oM.Geometry.ICurve externalBoundary = crvLoops.FirstOrDefault(x => x.IsCounterclockwise(face.FaceNormal))?.FromRevit();
+            List<oM.Geometry.ICurve> internalBoundary = crvLoops.Where(x => x != externalBoundary).Select(x => x.FromRevit() as oM.Geometry.ICurve).ToList();
```

Comment:
```suggestion
            CurveLoop outer = crvLoops.FirstOrDefault(x => x.IsCounterclockwise(face.FaceNormal));
oM.Geometry.ICurve externalBoundary = outer?.FromRevit();
            List<oM.Geometry.ICurve> internalBoundary = crvLoops.Where(x => x != outer).Select(x => x.FromRevit() as oM.Geometry.ICurve).ToList();
```

Reply:
Now I'm hyper-critical of anything you wrote! 🤣 
How about this?

```c#
        public static oM.Geometry.PlanarSurface FromRevit(this PlanarFace face)
        {
            if (face == null)
                return null;

            List<CurveLoop> crvLoops = face.GetEdgesAsCurveLoops().ToList();
            CurveLoop externalLoop = crvLoops.FirstOrDefault(x => x.IsCounterclockwise(face.FaceNormal));

            if (externalLoop == null)
            {
                BH.Engine.Base.Compute.RecordError($"A Revit face has no counter-clockwise boundary loop. Its conversion to BHoM ICurve may be inaccurate.");
                externalLoop = crvLoops[0];
            }

            oM.Geometry.ICurve externalBoundary = externalLoop.FromRevit();
            List<oM.Geometry.ICurve> internalBoundaries = crvLoops.Where(x => x != externalLoop).Select(x => x.FromRevit() as oM.Geometry.ICurve).ToList();

            return new oM.Geometry.PlanarSurface(externalBoundary, internalBoundaries);
        }
```

Reply:
I think I would return `null` in case of external loop not found - better to return nothing than a surface with wrong outline (I think the code would break further downstream anyways). So I would opt for something along the lines:
```C#
if (externalLoop == null)
{
    BH.Engine.Base.Compute.RecordError($"Conversion of element face to planar surface failed because face outline could not be found.");
    return null;
}
```